includes/class-wc-gateway-amazon-payments-advanced-subscriptions.php

```diff
@@ -408,6 +408,9 @@ public function scheduled_subscription_payment( $amount_to_charge, $order ) {
 		if ( is_wp_error( $response ) ) {
 			wc_apa()->log( "Error processing payment for renewal order #{$order_id}. Charge Permission ID: {$charge_permission_id}", $response );
 			$order->add_order_note( sprintf( __( 'Amazon Pay subscription renewal failed - %s', 'woocommerce-gateway-amazon-payments-advanced' ), $response->get_error_message() ) );
+			$charge_permission_status = wc_apa()->get_gateway()->log_charge_permission_status_change( $order );
+			$order->update_status( 'failed' );
+			wc_maybe_increase_stock_levels( $order->get_id() );
```

Comment:
This is a maybe function. This only reverts the decrease in case it already happened. Same for the success, you would find calls to the `wc_maybe_decrease_stock_levels` function.

Do you think this is not correct?

Reply:
Not so sure to say not correct, but do not looks consistent, like it should be taken care in other place and the decrease can be done twice this way. But I do not have so clear the process to be sure about this.